src/Symfony/Component/Messenger/Bridge/Amqp/Transport/Connection.php

```diff
@@ -351,41 +359,42 @@ private function getDelayExchange(): \AMQPExchange
 
     /**
      * Creates a delay queue that will delay for a certain amount of time.
-     *
-     * This works by setting message TTL for the delay and pointing
-     * the dead letter exchange to the original exchange. The result
-     * is that after the TTL, the message is sent to the dead-letter-exchange,
-     * which is the original exchange, resulting on it being put back into
-     * the original queue.
      */
-    private function createDelayQueue(int $delay, ?string $routingKey): \AMQPQueue
+    private function createDelayQueue(int $delay, ?string $originQueue, ?string $routingKey): \AMQPQueue
     {
         $queue = $this->amqpFactory->createQueue($this->channel());
-        $queue->setName(str_replace(
-            ['%delay%', '%exchange_name%', '%routing_key%'],
-            [$delay, $this->exchangeOptions['name'], $routingKey ?? ''],
-            $this->connectionOptions['delay']['queue_name_pattern']
-        ));
+
+        if (null === $originQueue) {
```

Comment:
I this pull request i wanted to split delay and retry logic. Delay logic requires original routing key. Retry logic doesn't. In both cases default exchange ('') can be used as the exchange to rote messages. I saved previous logic to make changes compatible with old configuration. As for me using of default exchange is more elegance, because you don't need to create "delay" exchange.

> the default exchange logic is now based on the X-Origin-Queue header
it's not exchange logic. It's `delay` or `retry` logic

> this header will always be set by the sender
not always. Only when message was received from normal queue. In case we want simly delay message, it's not  set.

> i think you want to make this based on useDefaultExchangeForDelay
i can't base on it, because i am not able to determine if `delay` or `retry` is required.